src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,49 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter);
         }
     };
 }
 
 using namespace RSS;
 
+QString computeEpisodeName(const QString &article)
+{
+    static const QRegularExpression episodeRegex(
+                    "(?:^|[^A-Za-z0-9])(?:"
+
+                    //Format 1: s01e01
+                    "(?:s(\\d{1,})e(\\d{1,}))|"
+
+                    //Format 2: 01x01
+                    "(?:(\\d{1,})x(\\d{1,}))|"
+
+                    //Format 3: 2017.01.01
+                    "(\\d{4}\\.\\d{2}\\.\\d{2})"
+
+                    ")(?:[^A-Za-z0-9]|$)",
```

Comment:
While using the + is probably better, using \W probably isn't, as I just want to make sure the season/episode number isn't matching within something else, and fields might be separated by '_'s.

As for adding more date formats, I'm not sure making the regex that lenient without at least some requirement to be dubious at best. And it also raises the question, should we interpret/reformat the date when we store it (like we do the episode numbers) so that we can continue to match if the format changes?